index.js

```diff
@@ -7,17 +7,23 @@ const EventEmitter = require('events')
 class HeadRush extends EventEmitter {
     constructor(eventsProvided) {
         super()
-        if (!events.length) events = eventsProvided
+        if (!events.length) {
+            events = eventsProvided
+        }
         if (!events && !eventsProvided) {
             throw new Error('Initial events need to be present to stun')
         }
     }
 
     stun(event) {
-        eventsFired[event] = true
-        if (Object.keys(eventsFired).length >= events.length) {
-            eventsFired = events = null
-            this.emit('ready')
+        if (events.indexOf(event) >= 0) {
+            eventsFired[event] = true
+            if (Object.keys(eventsFired).length >= events.length) {
+                eventsFired = events = null
+                this.emit('ready')
+            }
+        } else {
+            throw new Error('Can only stun initial events provided')
```

Comment:
Can we change this message to something more formal ? Something like Event stunned not present in the expected events set.